torch_xla/distributed/xla_backend.py

```diff
@@ -35,6 +35,10 @@ def __init__(self, prefix_store, rank, size, timeout):
     self.prefix_store = prefix_store  # reserved for future use.
     self.timeout = timeout
     self._mesh = []
+    if os.getenv('PJRT_DEVICE',
+                 None) == 'NEURON' and os.getenv('XLA_USE_SPMD') != '1':
+      dev = xm.xla_device()
+      xm.set_replication(dev, [dev])
```

Comment:
Do you normally call `dist.init_process_group` without `xmp.spawn`? That should already have set the replication. If it's not handling the SPMD case correctly, then I can follow up with a fix.

Reply:
Yeah we usually call it in torchrun cases as well

Reply:
Previously we had an xrt context function that would call the setting replication in this spot, we removed all of that since there's no xrt but we still needed the replication to be set

Reply:
`spawn` will have already called `set_replication` with all available devices in the current thread (usually just the one returned by `xm.xla_device`) by the time you reach this line: https://github.com/pytorch/xla/blob/f06c31f5685fd537e9f220f3fdae8116d0bd0c32/torch_xla/_internal/pjrt.py#L58-L59

Reply:
yeah this is mean to be for torchrun only - i can add a if torch.distributed.is_torchelastic_launched() here? Also, the only reason i added this only call if not GSPMD is because we don't have it enabled yet for neuron

Reply:
Got it. I just forgot that `xmp.spawn` doesn't behave correctly with torchrun. This is fine for now. I'm 90% sure you can remove `os.getenv('XLA_USE_SPMD') != '1'` with no issue here, even when you do get to enabling it.

Guarding this behind `dist.is_torchelastic_lauched` makes sense to me.

The logic still doesn't seem quite right to me, since it would make sense to initialize everything before you get to the PJRT rendezvous so that you can trust the `xm.world_size` there. I'll think through the problem some more and copy you on any changes I make to this logic.

Reply:
#5438 sets up multiprocessing in `init_process_group` when launched with torchrun and uses `torchrun`'s store instead of creating a new one in process 0. You should be able to remove this condition and the one in https://github.com/pytorch/xla/pull/5428#discussion_r1290526436.